draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -485,16 +485,29 @@ a Disclosure more than once.
 ### Enabling Holder Binding {#enabling_holder_binding}
 
 The Holder MAY add an optional JWT to prove Holder Binding to the Verifier.
-The precise contents of the JWT are out of scope of this specification.
-Usually, a `nonce` and `aud` claim are included to show that the proof is
-intended for the Verifier and to prevent replay attacks. How the `nonce` or
-other claims are obtained by the Holder is out of scope of this specification.
 
-Example Holder Binding JWT payload:
+#### Structure of the Holder Binding JWT {#hb-jwt}
+
+This section defines the contents of the Holder Binding JWT.
+
+The JWT MUST contain the following elements:
+  * in the JOSE header,
+    * `typ`: REQUIRED. MUST be `hb+jwt`, which explicitly types the Holder Binding JWT as recommended in Section 3.11 of [@!RFC8725].
+    * `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. MUST NOT be `none` or an identifier for a symmetric algorithm (MAC).
+    * `kid`: `kid`: CONDITIONAL. JOSE Header containing the key ID. MUST NOT be present if `jwk` or `x5c` is present.
+    * `jwk`: CONDITIONAL. JOSE Header containing the key material the new Credential shall be bound to. MUST NOT be present if `kid` or `x5c` is present.
+    * `x5c`: CONDITIONAL. JOSE Header containing a certificate or certificate chain corresponding to the key used to sign the JWT. This element MAY be used to convey a key attestation. In such a case, the actual key certificate will contain attributes related to the key properties. MUST NOT be present if `kid` or `jwk` is present.
```

Comment:
In general I think the JOSE headers that convey or point to the key don't need to be and shouldn't be in this list. The HB-JWT for an SD-JWT that is holder bound with a cnf.jwk doesn't need any of these, for one example. And the requirements re kid/jwk/x5c as written in these bullets now aren't totally clear (to me anyway) and need some fixup. Or just remove them.  I'd argue that the defined structure of the Holder Binding JWT should be only the minimal claims/headers necessary. Other requirements, if needed, such as saying that no more than one of `kid` or `jwk` or `x5c` be used together could be in text elsewhere.